src/base-button/base-button.ts

```diff
@@ -57,12 +58,22 @@ export abstract class BaseButton extends LitElement {
         aria-describedby=${nothing}
       >
         <span class="cover"></span>
-        <!-- TODO: add spinner -->
-        ${this.loading ? html`<span>loading</span>` : html` <slot></slot>`}
+        ${this.loading ? this.renderLoadingContent() : html` <slot></slot>`}
       </button>
     `;
   }
 
+  private renderLoadingContent = () => {
+    return html `
+      <div class="spinner">
```

Comment:
no need to wrap the `tap-spinner` in a `div`. we can use the class on the element itself.